de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/payment/JsonPaymentService.java

```diff
@@ -0,0 +1,137 @@
+
+	public JsonPaymentService(final CurrencyService currencyService, final BpartnerPriceListServicesFacade bpartnerPriceListServicesFacade)
+	{
+		this.currencyService = currencyService;
+		this.bpartnerPriceListServicesFacade = bpartnerPriceListServicesFacade;
+	}
+
+	public ResponseEntity<String> createPaymentFromJson(@NonNull @RequestBody final JsonPaymentInfo jsonPaymentInfo)
+	{
+		final LocalDate dateTrx = CoalesceUtil.coalesce(jsonPaymentInfo.getTransactionDate(), SystemTime.asLocalDate());
+
+		final CurrencyId currencyId = currencyService.getCurrencyId(jsonPaymentInfo.getCurrencyCode());
+		if (currencyId == null)
+		{
+			return ResponseEntity.unprocessableEntity().body("Wrong currency: " + jsonPaymentInfo.getCurrencyCode());
+		}
+
+		final OrgId orgId = retrieveOrg(jsonPaymentInfo);
+
+		final Optional<BPartnerId> bPartnerIdOptional = retrieveBPartnerId(IdentifierString.of(jsonPaymentInfo.getExternalBpartnerId()));
+		if (!bPartnerIdOptional.isPresent())
+		{
+			return ResponseEntity.unprocessableEntity().body("Cannot find bpartner: " + jsonPaymentInfo.getExternalBpartnerId());
+		}
+
+		final BPartnerId bPartnerId = bPartnerIdOptional.get();
+		final Optional<BankAccountId> bankAccountIdOptional = bankAccountDAO.retrieveBankAccountByBPartnerAndCurrencyAndIBAN(bPartnerId, currencyId, jsonPaymentInfo.getTargetIBAN());
+
+		if (!bankAccountIdOptional.isPresent())
+		{
+			return ResponseEntity.unprocessableEntity().body(String.format("Cannot find Bank Account for bpartner: %s, currency: %s and account: %s", jsonPaymentInfo.getExternalBpartnerId(), jsonPaymentInfo.getCurrencyCode(), jsonPaymentInfo.getTargetIBAN()));
+		}
+
+		final I_C_Payment payment = paymentBL.newOutboundPaymentBuilder()
+				.bpartnerId(bPartnerId)
+				.payAmt(jsonPaymentInfo.getAmount())
+				.currencyId(currencyId)
+				.bpBankAccountId(bankAccountIdOptional.get())
+
+				.adOrgId(orgId)
+				.tenderType(TenderType.DirectDeposit)
+				.dateAcct(dateTrx)
+				.dateTrx(dateTrx)
+				.createAndProcess();
+
+		final String externalOrderId = IdentifierString.of(jsonPaymentInfo.getExternalOrderId()).asExternalId().getValue();
+		payment.setExternalOrderId(externalOrderId);
+		InterfaceWrapperHelper.save(payment);
+
+		return ResponseEntity.ok().build();
+	}
+
+	private OrgId retrieveOrg(@RequestBody @NonNull final JsonPaymentInfo jsonPaymentInfo)
+	{
+		final Optional<OrgId> orgId;
+		if (Check.isNotBlank(jsonPaymentInfo.getOrgCode()))
+		{
+			final OrgQuery query = OrgQuery.builder()
+					.orgValue(jsonPaymentInfo.getOrgCode())
+					.build();
+			orgId = orgDAO.retrieveOrgIdBy(query);
+		}
+		else
+		{
+			orgId = Optional.empty();
+		}
+		return orgId.orElse(Env.getOrgId());
+	}
+
+	private Optional<BPartnerId> retrieveBPartnerId(final IdentifierString bPartnerIdentifierString)
+	{
+		// TODO it would be nice here if we would also use orgID when searching for the bpartner, since an ExternalId may belong to multiple users from different orgs.
```

Comment:
ok for now. if you meet this piese of code abain, place extract the code from bpartnerPriceListServicesFacade into a dedicated class

Reply:
![image](https://user-images.githubusercontent.com/4482210/70239991-0f170200-1775-11ea-8679-56a1af3057e6.png)